### ISSUE TITLE
Non-integer indices raise SystemError rather than TypeError on assignment

### ISSUE THREAD

**Issue by cycloid** [ilanschnell/bitarray#114]
If a non-integer index value is passed on an assignment a SystemError is raised rather than the expected TypeError.

Python: 3.9.2 (Anaconda)
Bitarray: 1.7.0
MacOS 10.15.7/Intel

```
Python 3.9.2 (default, Mar  3 2021, 11:58:52)
[Clang 10.0.0 ] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from bitarray.util import zeros
>>> b = zeros(20)
>>> b
bitarray('00000000000000000000')
>>> b[1.5]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: index or slice expected
>>> b[1.5] = 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
SystemError: error return without exception set
>>> b
bitarray('00000000000000000000')
```

**Comment by ilanschnell** [comment]
Thank you for reporting this issue.  I just fixed the problem on the master branch.  This fix will be part of the 1.7.1 release.

**Comment by cycloid** [comment]
Thank you for the quick response!

**Comment by ilanschnell** [comment]
I just released version 1.7.1 which includes this fix.
